fetch.js

```diff
@@ -424,93 +408,32 @@ Response.redirect = function (url, status) {
   return new Response(null, {status: status, headers: {location: url}})
 }
 
-export var DOMException = self.DOMException
-try {
-  new DOMException()
-} catch (err) {
-  DOMException = function (message, name) {
-    this.message = message
-    this.name = name
-    var error = Error(message)
-    this.stack = error.stack
-  }
-  DOMException.prototype = Object.create(Error.prototype)
-  DOMException.prototype.constructor = DOMException
+var defaults = {
+  // To replicate the default browser fetch behavior.
+  muteHttpExceptions: true,
 }
 
 export function fetch(input, init) {
-  return new Promise(function (resolve, reject) {
-    var request = new Request(input, init)
-
-    if (request.signal && request.signal.aborted) {
-      return reject(new DOMException('Aborted', 'AbortError'))
-    }
-
-    var xhr = new XMLHttpRequest()
-
-    function abortXhr() {
-      xhr.abort()
-    }
-
-    xhr.onload = function () {
-      var options = {
-        status: xhr.status,
-        statusText: xhr.statusText,
-        headers: parseHeaders(xhr.getAllResponseHeaders() || ''),
-      }
-      options.url = 'responseURL' in xhr ? xhr.responseURL : options.headers.get('X-Request-URL')
-      var body = 'response' in xhr ? xhr.response : xhr.responseText
-      resolve(new Response(body, options))
-    }
-
-    xhr.onerror = function () {
-      reject(new TypeError('Network request failed'))
-    }
-
-    xhr.ontimeout = function () {
-      reject(new TypeError('Network request failed'))
-    }
-
-    xhr.onabort = function () {
-      reject(new DOMException('Aborted', 'AbortError'))
-    }
-
-    xhr.open(request.method, request.url, true)
-
-    if (request.credentials === 'include') {
-      xhr.withCredentials = true
-    } else if (request.credentials === 'omit') {
-      xhr.withCredentials = false
-    }
-
-    if ('responseType' in xhr && support.blob) {
-      xhr.responseType = 'blob'
+  return Promise.resolve().then(function () {
+    if (init === undefined || init.muteHttpExceptions == null) {
+      init = Object.assign({}, defaults, init)
     }
 
-    request.headers.forEach(function (value, name) {
-      xhr.setRequestHeader(name, value)
-    })
+    var urlFetchAppRequest = UrlFetchApp.fetch(input, init)
```

Comment:
What is `UrlFetchApp`? I see that it is a global object, wouldn't it be better if it was imported instead?

Reply:
Got it, https://developers.google.com/apps-script/reference/url-fetch/url-fetch-app. LGTM now.

Reply:
:)

Reply:
I always miss checking the description in PRs as well :)